internal/abi/fastly/hostcalls_guest.go

```diff
@@ -2436,8 +2436,13 @@ func fastlyDictionaryOpen(
 // Dictionary represents a Fastly edge dictionary, a collection of read-only
 // key/value pairs. For convenience, keys are modeled as Go strings, and values
 // as byte slices.
+//
+// NOTE: wasm, by definition, is a single-threaded execution environment. This
```

Comment:
While wasm is single-threaded, both TinyGo and Big Go have support for goroutines via their own preemption strategies.  While we can probably get away with a single buffer, I still think it should be protected by a mutex.

Reply:
Completely reasonable